austrakka/components/user/funcs.py

```diff
@@ -35,23 +35,49 @@ def add_user(
     )
 
 
-@logger_wraps()
 def update_user(
-    user_id: str,
-    org: str,
-    owner_group_roles: List[str],
-    is_austrakka_process: bool,
+        object_id: str,
+        name: str = None,
+        email: str = None,
+        org: str = None,
+        group_roles: List[str] = None,
+        is_active: bool = None,
 ):
-    user = {
-        "organisation": {
-            "abbreviation": org
-        },
-        "ownerGroupRoles": list(owner_group_roles),
-        "isAusTrakkaProcess": is_austrakka_process,
+    user_resp = api_get(f'{USER_PATH}/userId/{object_id}')
+    user_full = user_resp['data']
+    user: Dict[str, Any] = {
+        "displayName": user_full['displayName'],
+        "contactEmail": user_full['contactEmail'],
+        "orgAbbrev": user_full['orgAbbrev'],
+        "groupRoles": [
+            {
+                "roleName": role_dto["role"]["name"],
+                "groupName": role_dto["group"]["name"],
+            }
+            for role_dto in user_full['groupRoles']
+        ],
+        "isActive": user_full['isActive'],
     }
 
+    if name is not None:
+        user['displayName'] = name
+    if email is not None:
+        user['contactEmail'] = email
+    if org is not None:
+        user['orgAbbrev'] = org
+    if group_roles is not None:
```

Comment:
This is good (taking the current state and updating it with supplied properties), I'd just say drop the group roles update as per the above comment.